src/libraries/Native/Unix/System.Security.Cryptography.Native.Android/pal_x509.c

```diff
@@ -287,3 +295,52 @@ static int32_t PopulateByteArray(JNIEnv* env, jbyteArray source, uint8_t* dest,
     (*env)->GetByteArrayRegion(env, source, 0, bytesLen, (jbyte*)dest);
     return CheckJNIExceptions(env) ? FAIL : SUCCESS;
 }
+
+static void FindCertStart(const uint8_t** buffer, int32_t* len)
+{
+    if (iscntrl(**buffer) && !isspace(**buffer))
+    {
+        // If the character is a control character that isn't whitespace, then we're probably using a DER encoding
+        // and not using a PEM encoding in ASCII.
+        return;
+    }
+    assert(buffer != NULL && *buffer != NULL);
+    assert(*len >= 0);
```

Comment:
The asserts seem like they should be before the first deferences :)

Reply:
Yes they should :(
I added the first check after I wrote the assert.